pkg/policy/resolve.go

```diff
@@ -165,13 +162,15 @@ func (p *EndpointPolicy) computeDirectionL4PolicyMapEntries(l4PolicyMap L4Policy
 	for _, filter := range l4PolicyMap {
 		lookupDone := false
 		proxyport := uint16(0)
-		keysFromFilter := filter.ToMapState(p.NamedPortsMap, direction)
+		keysFromFilter := filter.ToMapState(p.PolicyOwner, direction)
 		for keyFromFilter, entry := range keysFromFilter {
 			// Fix up the proxy port for entries that need proxy redirection
 			if entry.IsRedirectEntry() {
-				if !lookupDone {
+				if !lookupDone && filter.IsRedirect() {
```

Comment:
What's the implication of this extra check here? The entry is from `filter.ToMapState()`, so the filter must have a redirect otherwise presumably `entry.isRedirectEntry()` would return false ..?

Reply:
This was a result of refactoring a corresponding check out of `LookupRedirectPortLocked()` (called below), but you are right, it is not needed due to filter's entry already being checked on the previous line. Removed :-)